GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -736,7 +744,7 @@ private void RegisterPlugins()
                         if (plugin.Execute(new GitUIEventArgs(this, UICommands)))
                         {
                             _gitStatusMonitor.InvalidateGitWorkingDirectoryStatus();
-                            UICommands.RepoChangedNotifier.Notify();
+                            RefreshRevisions();
```

Comment:
A drawback of this might be that a future subscriber of `UICommands.RepoChangedNotifier` might not get updated.

Reply:
If the repo is changed, they will.
If a notification is required for other reasons, there should be a separate notification (like the existing GridLoading).
The intention of this was to make the handling possible to understand.